server/proxy/tcp.go

```diff
@@ -130,12 +132,18 @@ func registerClient(apiKey string, port int, endpoints string, shouldDelete bool
 		url := fmt.Sprintf("%s/ng/api/tunnel?accountIdentifier=%s", endPoint, accoundId)
 		contentType := "application/json"
 
-		hasher := md5.New()
-		hasher.Write([]byte(apiKey))
-		hashBytes := hasher.Sum(nil)
-		hashString := hex.EncodeToString(hashBytes)
+		var uniqueSha string
+		if harnessUsername != "" && harnessPassword != "" {
+			uniqueSha = fmt.Sprintf("%s:%s", harnessUsername, harnessPassword)
+		} else {
```

Comment:
why this would be empty if you set it on the config level ? you can delete what you do on the convert level and keep this here or keep it there and throw and error if pwd and uname is empty here.

Reply:
No need to throw an error. If the user did not specify username/password, we'll use the API token.